packages/block-library/src/navigation/index.php

```diff
@@ -5,148 +5,6 @@
-	 * The final HTML of the navigation-submenu and the page-list blocks will look similar to this:
-	 *
-	 * <li
-	 *   class="has-child"
-	 *   data-wp-context='{ "core": { "navigation": { "isMenuOpen": false, "overlay": false } } }'
-	 *   data-wp-effect="effects.core.navigation.initMenu"
-	 *   data-wp-on.keydown="actions.core.navigation.handleMenuKeydown"
-	 *   data-wp-on.focusout="actions.core.navigation.handleMenuFocusout"
-	 * >
-	 *   <button
-	 *     class="wp-block-navigation-submenu__toggle"
-	 *     data-wp-on.click="actions.core.navigation.openMenu"
-	 *     data-wp-bind.aria-expanded="context.core.navigation.isMenuOpen"
-	 *   >
-	 *   </button>
-	 *   <span>Title</span>
-	 *   <ul class="wp-block-navigation__submenu-container">
-	 *     SUBMENU ITEMS
-	 *   </ul>
-	 * </li>
-	 *
-	 * @param string $w Markup of the navigation block.
-	 * @param array  $block_attributes Block attributes.
-	 *
-	 * @return string Submenu markup with the directives injected.
-	 */
-	function block_core_navigation_add_directives_to_submenu( $w, $block_attributes ) {
-		while ( $w->next_tag(
-			array(
-				'tag_name'   => 'LI',
-				'class_name' => 'has-child',
-			)
-		) ) {
-			// Add directives to the parent `<li>`.
-			$w->set_attribute( 'data-wp-interactive', true );
-			$w->set_attribute( 'data-wp-context', '{ "core": { "navigation": { "isMenuOpen": { "click": false, "hover": false }, "overlay": false } } }' );
-			$w->set_attribute( 'data-wp-effect', 'effects.core.navigation.initMenu' );
-			$w->set_attribute( 'data-wp-on--focusout', 'actions.core.navigation.handleMenuFocusout' );
-			$w->set_attribute( 'data-wp-on--keydown', 'actions.core.navigation.handleMenuKeydown' );
-			if ( ! isset( $block_attributes['openSubmenusOnClick'] ) || false === $block_attributes['openSubmenusOnClick'] ) {
-				$w->set_attribute( 'data-wp-on--mouseenter', 'actions.core.navigation.openMenuOnHover' );
-				$w->set_attribute( 'data-wp-on--mouseleave', 'actions.core.navigation.closeMenuOnHover' );
-			}
-
-			// Add directives to the toggle submenu button.
-			if ( $w->next_tag(
-				array(
-					'tag_name'   => 'BUTTON',
-					'class_name' => 'wp-block-navigation-submenu__toggle',
-				)
-			) ) {
-				$w->set_attribute( 'data-wp-on--click', 'actions.core.navigation.toggleMenuOnClick' );
-				$w->set_attribute( 'data-wp-bind--aria-expanded', 'selectors.core.navigation.isMenuOpen' );
-			};
-
-			// Iterate through subitems if exist.
-			block_core_navigation_add_directives_to_submenu( $w, $block_attributes );
-		}
-		return $w->get_updated_html();
-	};
-
-	/**
-	 * Replaces view script for the Navigation block with version using Interactivity API.
-	 *
-	 * @param array $metadata Block metadata as read in via block.json.
-	 *
-	 * @return array Filtered block type metadata.
-	 */
-	function gutenberg_block_core_navigation_update_interactive_view_script( $metadata ) {
```

Comment:
I'm not sure they are removed in PHP: https://github.com/WordPress/wordpress-develop/blob/1815f4c76c9f51665f046d79284552d09849597b/src/wp-includes/blocks/file.php#L8

In JS, yes.

I just threw this PR up as a "just in case". We could easily rename the functions too, but keep them in a wrapper.

Reply:
> I'm not sure they are removed in PHP

Sorry, that link was to an unbuilt file. I just checked my build folder locally and can still see the `IS_GUTENBERG_PLUGIN` blocks.

Reply:
Yeah, the code is there, but no used in core. Check my [comment](https://github.com/WordPress/gutenberg/issues/52103#issuecomment-1619860446) in the issue.

>The blocks code is built automatically in WP core folder structure. There are some features/code that needs to be part of the php blocks' code, but we don't want it to be part of WP core.

So we should remove the code only if the code is not used/needed anymore.

Reply:
> So we should remove the code only if the code is not used/needed anymore.

Ah, okay. No worries. Just to confirm, the code is needed in the plugin but not in Core. The interactivity API will not be backported to WP 6.3.

So better to rename the functions? It's not quite the same as in https://github.com/WordPress/gutenberg/pull/51989, which only renamed to an existing core function that had been around for a while. Here we are potentially introducing two new, currently experimental, functions, albeit guarded by the `IS_GUTENBERG_PLUGIN` checks. 

As long as we are sure this is a good approach, then I can rejig this PR.

Reply:
>It's not quite the same as in https://github.com/WordPress/gutenberg/pull/51989,

That's a different use case. Since the code in blocks.php files are copied for core, we can't have functions that only exist in GB. Additionally, if blocks call some core functions(like the above PR) and we want to override in GB, we have the Webpack [mechanism](https://github.com/WordPress/gutenberg/pull/51989/files#diff-777759e98c224019a699e8294c23bccc34453c91b873a12cb78a828487cd5f15R30) to add the `gutenberg` prefix to these specific functions.

Also I just realized you're targeting the `6.3` branch. 🤔 I'll cc @Mamaduka too

Reply:
I think renaming the functions in trunk is the more future-proof approach. If all goes well, at some point the logic will make it into core and they'll have to be renamed all the same, so we might as well do it straightaway. In any case, given the block PHP files are auto-generated in core, it's good practice to never use `gutenberg` prefixes in them. If we need the functions to be namespaced for development reasons, we can use [the webpack task](https://github.com/WordPress/gutenberg/blob/trunk/tools/webpack/blocks.js#L30C26-L30C26) to add the prefix at build time.

I also don't love the idea of introducing changes directly to the release branch, as it can cause conflicts if we later need to make changes to the files. This branch will still be used for minor releases after 6.3.0 is out.

Reply:
Thanks for the feedback folks. I'll update this PR and change the base branch to trunk.

Reply:
Actually, I'll create a new branch. This one has a name that might confuse: `remove/is-gutenberg-plugin-from-6-3`